### DISCUSSION TITLE
Link header based pagination

### DISCUSSION THREAD

**Discussion by tanoc** [vitalik/django-ninja#1284] [Q&A]
Is there a way to support link header based pagination (something like [this](https://docs.github.com/en/rest/using-the-rest-api/using-pagination-in-the-rest-api?apiVersion=2022-11-28#using-link-headers)) using a custom pagination class? I'm probably missing something but I'm not seeing how to change response headers from there.

**Comment by vitalik** [comment]
@tanoc 

I guess you can make some decorator that will add `link` header


```Python
from ninja.pagination import paginate
from ninja.decorators import decorate_view


def add_link_header(func):
     def wrapper(request, *a, **kw):
          response = func(request, *a, **kw)
          response.headers['link'] = 'xxxx' # <-- TODO
          return response
     return wrapper

@api.get('/some')
@decorate_view(add_link_header) # <--- make sure decorate on view layer
@paginate
def some(request):
      return queryset
```

**Reply by tanoc** [reply]
Thanks, but I would still need to be able to access the paginator object to know for example the page number (to set for example previous and next page links). I could probably attach the custom paginator to the request object, but it would be kinda ugly and I'm not sure how stable it will be.

Maybe having access to the response object (like [this](https://github.com/vitalik/django-ninja/pull/336)) in the paginator object (`paginate_queryset`) could be an idea?

**Reply by vitalik** [reply]
I do not yet see way to access response in paginate_quryset...

but in decorator you can:

```Python
def add_link_header(func):
     def wrapper(request, *a, **kw):
          response = func(request, *a, **kw)
          page = request.GET['page'] # <-------------------
          response.headers['link'] = 'xxxx'
          return response
     return wrapper
```

**Reply by tanoc** [reply]
yes, that's what I thought. but maybe it could be useful to be able to access it in the next versions. and it's not just the page number, but I would probably need info from the paginator like knowing if there is a next page or not.
